From f625d4cb52d9484b7b94607ba1a0b36ca8ce64d4 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 09:48:03 +0530
Subject: [PATCH 007/500] OpenWrt: 142-jffs2-add-splice-ops.patch

jffs2: add splice ops

Add splice_read using generic_file_splice_read.
Add splice_write using iter_file_splice_write

Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: Id5dbbfa5315d08cc6bd37f133528d85c3c0e8b96
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit b005bcc2dc24440dcc5f6b718aa30d4fb5ea064a)
---
 fs/jffs2/file.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/jffs2/file.c b/fs/jffs2/file.c
index 11c66793960e..fbc3297142bc 100644
--- a/fs/jffs2/file.c
+++ b/fs/jffs2/file.c
@@ -53,6 +53,8 @@ const struct file_operations jffs2_file_operations =
 	.open =		generic_file_open,
  	.read_iter =	generic_file_read_iter,
  	.write_iter =	generic_file_write_iter,
+	.splice_read =  filemap_splice_read,
+	.splice_write =	iter_file_splice_write,
 	.unlocked_ioctl=jffs2_ioctl,
 	.mmap =		generic_file_readonly_mmap,
 	.fsync =	jffs2_fsync,
-- 
2.34.1

